### PR TITLE
[RW-7050][risk=no] rm Angular signed in, disabled guards

### PR DESCRIPTION
**Pull request by calbach**
- Sign in guard is already enforced within the React router
- Disabled guard is now replaced by a new React guard, this uses a new custom hook introduced to track the user's disabled status

### CODE REVIEW

File: ui/src/app/app-routing.module.ts [hunk 1]
```diff
@@ -37,342 +35,325 @@ const routes: Routes = [
...
+  exports: [RouterModule]
 })
 export class AppRoutingModule {
 
-  constructor(public router: Router) {
-    NavStore.navigate = (commands, extras) => this.router.navigate(commands, extras);
-    NavStore.navigateByUrl = (url, extras) => this.router.navigateByUrl(url, extras);
+  constructor(private router: Router, private zone: NgZone) {
+    NavStore.navigate = (commands, extras) => {
+      zone.run(() => this.router.navigate(commands, extras));
```
Comment (calbach) [inline ui/src/app/app-routing.module.ts:352]: Found this fix after hours of debugging..

File: ui/src/app/app-routing.module.ts [hunk 2]
```diff
@@ -37,342 +35,325 @@ const routes: Routes = [
     path: 'user-disabled',
     component: AppRouting
   },
-  {
```
Comment (calbach) [inline ui/src/app/app-routing.module.ts:40]: Changes here are minor, hide whitespace to see.

**Review by als364 — COMMENTED** [review]
While we do have a Reacty signed-in guard, I don't think we currently have anything replacing DisabledGuard.

**Review by calbach — COMMENTED** [review]
Ooops, meant to send this yesterday.

Puppeteer failures seem like they may be legitimate, will look later today.

File: ui/src/app/pages/workspace/workspace-about.tsx [hunk 3]
```diff
@@ -109,17 +109,17 @@ export const WorkspaceAbout = fp.flow(withUserProfile(), withUrlParams(), withCd
     };
   }
 
-  async componentDidMount() {
+  componentDidMount() {
```
Comment (calbach) [inline ui/src/app/pages/workspace/workspace-about.tsx:112]: This was the source of most of the puppeteer failures. I'm surprised that this was working previously - but it was definitely highly dependent on arbitrary React rendering order.

Specifically, it was breaking the rules of React state:

```
this.setState({workspace});
...
// expecting this.state to have been synchronously updated
f(this.state.workspace);
```

The other changes in the PR must have somehow resulted in an ordering change, possibly this was always a risk but a race condition was removed.

File: ui/src/app/app-routing.module.ts [hunk 4]
```diff
@@ -36,343 +34,323 @@ const routes: Routes = [
...
-                    path: 'duplicate',
+                    path: 'preview/:nbName',
+                    component: AppRouting,
+                    data: {}
+                  }
+                ]
+              },
+              {
+                path: 'data',
+                canDeactivate: [NavigationGuard],
```
[the remark this answers is not in the document]
Comment (calbach) [reply ui/src/app/app-routing.module.ts]: I moved it up one level, to where "runguardsandresolvers" was defined. It actually has a dependency on this; that was the source of the cohort builder puppeteer failure.

File: ui/src/app/app-routing.module.ts [hunk 5]
```diff
@@ -36,343 +34,325 @@ const routes: Routes = [
...
+            path: 'build',
+            component: AppRouting,
+            data: {}
+          },
+          {
+            /* TODO The children under ./views need refactoring to use the data
+             * provided by the route rather than double-requesting it.
+             */
+            path: ':ns/:wsid',
+            // TODO: When removing ng router, replace this with react-router's Prompt component.
```
Comment (als364) [inline ui/src/app/app-routing.module.ts:97]: Could you please either create a ticket for tearing out the angular router or attach this TODO to RW-1920, which is likely where we'd have to tear this out anyways?

**Review by als364 — APPROVED** [review]
One last nit

File: ui/src/app/app-routing.module.ts (hunk 5 above)
Comment (calbach) [reply ui/src/app/app-routing.module.ts]: Thanks, good call. This is in our style guide now after all.

**Pull request merged**